e2e/tests/batched_auction_data_reader_test.rs

```diff
@@ -0,0 +1,61 @@
+use driver::orderbook::{PaginatedStableXOrderBookReader, StableXOrderBookReading};
+use e2e::common::FutureBuilderExt;
+use e2e::stablex::setup_stablex;
+use ethcontract::web3::api::Web3;
+use ethcontract::web3::transports::Http;
+use ethcontract::Account;
+
+#[test]
+fn test_with_ganache() {
```

Comment:
To me, this could be a unit test with mocked contract responses. Alternatively, you could set `AUCTION_DATA_BATCH_SIZE=5` and setup an auction with many orders and check that it is solved.